pkg/chunkenc/dumb_chunk.go

```diff
@@ -71,7 +70,7 @@ func (c *dumbChunk) Encoding() Encoding { return EncNone }
 
 // Returns an iterator that goes from _most_ recent to _least_ recent (ie,
 // backwards).
-func (c *dumbChunk) Iterator(_ context.Context, from, through time.Time, direction logproto.Direction, _ log.StreamPipeline) (iter.EntryIterator, error) {
+func (c *dumbChunk) Iterator(_ StatsContext, from, through time.Time, direction logproto.Direction, _ log.StreamPipeline) (iter.EntryIterator, error) {
```

Comment:
How are `Context` and `StatsContext` different?

Reply:
Q: I see these are exported methods. Can the caller still be able to convert normal `context` for `StatsContext`? (if `context` is all they have).

Reply:
Yes that how it is used,you created a statscontext from a context. This allow to avoid getting the statscontext from context everything and avoid allocations.